webhook_server/tests/test_prepare_retest_welcome_comment.py

```
@@ -95,3 +104,226 @@ def test_prepare_retest_welcome_comment(
+
+
+class TestWelcomeMessageNewlineStructure:
+    """Regression tests for welcome message markdown formatting.
+
+    These tests ensure that conditional sections in the welcome message
+    maintain proper newline structure to prevent headers from being
+    glued to previous content.
+    """
+
+    def _create_handler(
+        self, process_github_webhook: GithubWebhook, owners_file_handler: OwnersFileHandler, **config: Any
```

Ruff (0.14.10)

Dynamically typed expressions (typing.Any) are disallowed in `**config`

(ANN401)